cypher-shell/src/main/java/org/neo4j/shell/state/BoltStateHandler.java

```diff
@@ -184,9 +183,17 @@ private void reconnect(boolean keepBookmark) {
 
         resetActualDbName(); // Set this to null first in case run throws an exception
         StatementResult run = session.run(query);
-        ResultSummary summary = run.consume();
-        this.version = summary.server().version();
-        updateActualDbName(summary);
+        ResultSummary summary = null;
+        try {
+            summary = run.consume();
+        } finally {
+            // Since run.consume() can throw the first time we have to go through this extra hoop to get the summary
```

Comment:
Why? Is this known/documented behavior?

Reply:
Hmm, it throws because it is not authorized to execute the ping query when password change is required in 4.0 anymore. It just started happening when packaged with the very latest Neo4j, but I wonder wether this really changed that recently. Maybe the neo4j docker image that is used to run integration tests is too far behind or there is some problem with the Team City test config parameters.

Reply:
@pontusmelke Ideally we want to get the version and database name by the simplest possible Bolt message or command that is always allowed to run (at least if authenticated), instead of having to execute an actual query, but this should do until we get Neo4j server support and/or driver support.